superset/utils/core.py

```diff
@@ -285,17 +284,33 @@ def decode_dashboards(o):
         return o
 
 
-class DashboardEncoder(json.JSONEncoder):
+class DashboardEncoder():
+
     # pylint: disable=E0202
-    def default(self, o):
+    @classmethod
+    def encode(cls, o):
         try:
-            vals = {
-                k: v for k, v in o.__dict__.items() if k != '_sa_instance_state'}
-            return {'__{}__'.format(o.__class__.__name__): vals}
-        except Exception:
-            if type(o) == datetime:
+            if isinstance(o, uuid.UUID):
+                return str(o)
+            if isinstance(o, datetime):
                 return {'__datetime__': o.replace(microsecond=0).isoformat()}
-            return json.JSONEncoder.default(self, o)
+            if isinstance(o, list):
+                return [DashboardEncoder.encode(i) for i in o]
+            if hasattr(o, '__dict__'):
+                vals = {}
+                for k, v in o.__dict__.items():
+                    if k == '_sa_instance_state':
+                        continue
+                    elif k.startswith('json') or k.endswith('json'):
+                        vals[k] = v
+                    else:
+                        vals[k] = DashboardEncoder.encode(v)
+                return {'__{}__'.format(o.__class__.__name__): vals}
+            else:
+                return o
+        except Exception as e:
+            logging.exception(e)
+            return o
```

Comment:
Hmmm, log and return. Do we know what may be happening here? Wishful error processing?

Reply:
We could not handle this at all. Previously an exception was thrown each and every time a `datetime.datetime` was detected. It is probably best that I just not handle the exception and let it die.

Reply:
Fixed.